extension/app/ts/background/backgroundUtils.ts

```diff
@@ -19,3 +19,7 @@ export async function sendPopupMessageToOpenWindows(message: MessageToPopup) {
 		}
 	}
 }
+
+export async function sendPopupMessageToBackgroundPage(message: PopupMessage) {
+	await browser.runtime.sendMessage(PopupMessage.serialize(message))
+}
```

Comment:
At a glance, it appears this could be made easier to use by changing the parameters:
```suggestion
export async function sendPopupMessageToBackgroundPage(method: PopupMessage['method'], options: PopupMessage['options']) {
	await browser.runtime.sendMessage(PopupMessage.serialize({ method, options }))
}
```
This would make it so the caller didn't have to repetitiously type so much and could just do `sendPopupMesageToBackgroundPage('popup_removeAddressBookEntry', { address, addressBookCategory })`.

Reply:
not all Popup messages have options field, so I am not sure how we could get this working with that. On option would be to add empty options on all such messages

Reply:
hmm, then the challenge is that you always need to match the method and options types to each other, I am not sure typescript can do that. Maybe with templates?

Reply:
Ah, I didn't realize `PopupMessage` was a union.  Do this instead then:
```suggestion
export async function sendPopupMessageToBackgroundPage<T extends PopupMessage>(method: T['method'], options: T['options']) {
	await browser.runtime.sendMessage({ method, options })
}
```
I *think* that will handle optional `options`, though I'm not 100% certain.